lib/keycloak.rb

```diff
@@ -210,14 +210,16 @@ def self.get_token_introspection(token = '', client_id = '', secret = '', intros
       exec_request _request
     end
 
-    def self.url_login_redirect(redirect_uri, response_type = 'code', client_id = '', authorization_endpoint = '')
+    def self.url_login_redirect(redirect_uri, response_type = 'code', client_id = '', authorization_endpoint = '', scope: nil)
       verify_setup
 
       client_id = @client_id if isempty?(client_id)
       authorization_endpoint = @configuration['authorization_endpoint'] if isempty?(authorization_endpoint)
 
-      p = URI.encode_www_form(response_type: response_type, client_id: client_id, redirect_uri: redirect_uri)
-      "#{authorization_endpoint}?#{p}"
+      params = { response_type: response_type, client_id: client_id, redirect_uri: redirect_uri }
+      params[:scope] = scope if scope
+      encoded_uri = URI.encode_www_form(params)
```

Comment:
if its an array like [email, openid], please make sure the url contains like this 

`&scope=email,openid`

and not like this **&scope=email&scope=openid**

Reply:
Thanks for the comment @ssa-wtag vai,

I made the changes now scope is an array and it will separate the items with comma, please see the below snapshot,

![Screenshot 2024-05-28 at 2 18 28 PM](https://github.com/Garaio-REM/keycloak/assets/57310683/a21cb8e6-5939-4ec2-81a5-19b0214116f9)

Also closing the MR due to wrong branch name, can you please have a look into this new MR https://github.com/Garaio-REM/keycloak/pull/2